scripts/build/build/target.py

```diff
@@ -264,6 +285,44 @@ def HumanString(self):
 
         return result
 
+    def ToDict(self):
+        """Outputs a parseable description of the available variants
+        and modifiers:
+
+            like:
+
+            {
+                "name": "foo"
+                "shorthand": "foo-bar-baz[-m1]"
+                "parts": [
+                    {
+                        "name": "foo",
+                        "build_arguments": {
+                            "board": "bar"
+                        }
+                    }
+                    {
+                        "name": "baz",
+                        "build_arguments": {
+                            "app": "foo.baz"
+                        }
+                    }
+                ],
+                "modifiers": [
+                    {
+                        "name": "modifier1",
+                        "m1": "True"
+                    }
+                ]
+            }
+        """
+        return {
+            'name': self.name,
+            'shorthand': self.HumanString(),
+            'parts': [part.ToDict() for target in self.fixed_targets for part in target],
```

Comment:
we may want an extra level of nesting here since this flattens parts. like `[[part.toDict() for part in target] for target in self.fixed_targets]`